src/ProcessManager/ProcessManager.php

```diff
@@ -21,15 +22,26 @@ final class ProcessManager implements ProcessManagerInterface
      */
     private $maxParallelProcesses;
 
+    /**
+     * @var IOInterface the IO interface
+     */
+    private $io;
+
     /**
      * @var Process[]
      */
-    private $processes;
+    private $processes = [];
+
+    /**
+     * @var int the operations to be processed
+     */
+    private $operations = 0;
 
-    public function __construct($timestamp, $maxParallelProcesses)
+    public function __construct($timestamp, $maxParallelProcesses, IOInterface $io)
```

Comment:
Here you use timestamp instead of timeout, it is less confusing

Reply:
yes, but it's not a timestamp... I rename it to "frequency" ?

Reply:
yep same as in the other comment, frequency is fine